nttt/utilities.py

```diff
@@ -55,4 +57,54 @@ def find_files(src, file_names=[], extensions=[]):
             if valid_file:
                 files_found.append(fpath)
 
-    return files_found
\ No newline at end of file
+    return files_found
+
+
+#PUB-9 utilities. These methods are for trimming spaces on specific markdown like ** or _. The main method is trim_spaces_on_specific_markdown which uses trim_spaces_on_specific_markdown_breakdown and strings_with_specific_markdown_and_spaces.
+ 
+
+regular_expressions_matrix_for_trimming_spaces = [
+    {'character':'_', 'regular_expression_for_wrong_left':r"_[ ]+[a-z]+[ ]*_", 'regular_expression_for_wrong_right':r"_[ ]*[a-z]+[ ]+_", "regular_expression_for_correct":r"_[a-z]*_"},
+    {'character':'`', 'regular_expression_for_wrong_left':r"`[ ]+[a-z]+[ ]*`", 'regular_expression_for_wrong_right':r"`[ ]*[a-z]+[ ]+`", "regular_expression_for_correct":r"`[a-z]*`"},
+    {'character':'<asterisk>', 'regular_expression_for_wrong_left':r"<asterisk>[ ]+[a-z]+[ ]*<asterisk>", 'regular_expression_for_wrong_right':r"<asterisk>[ ]*[a-z]+[ ]+<asterisk>", "regular_expression_for_correct":r"<asterisk>[a-z]*<asterisk>"},
+    {'character':'*', 'regular_expression_for_wrong_left':r"\*[ ]+[a-z]+[ ]*\*", 'regular_expression_for_wrong_right':r"\*[ ]*[a-z]+[ ]+\*", "regular_expression_for_correct":r"\*[a-z]*\*"},
+    ]
+
+corrections_for_trimming_spaces = [
+    {'character':'_', 'left_space':"_ ", 'right_space':" _"},
+    {'character':'`', 'left_space':"` ", 'right_space':" `"},
+    {'character':'<asterisk>', 'left_space':"<asterisk> ", 'right_space':" <asterisk>"},
+    {'character':'*', 'left_space':"* ", 'right_space':" *"},
+    ]
+
+def trim_spaces_on_specific_markdown(content):
+    content = content.replace("**", "<asterisk>")
+    for i in [0, 1, 2, 3]:
```

Comment:
`range` function would be useful here: https://www.w3schools.com/python/python_for_loops.asp